shared/modules/hexstring-utils.js

```diff
@@ -73,3 +73,19 @@ export function toChecksumHexAddress(address) {
   }
   return toChecksumAddress(hexPrefixed);
 }
+
+export function isHexPrefixed(str) {
+  if (typeof str !== 'string') {
+    throw new Error(
+      `Value must be type 'string', is currently type ${typeof str}, while checking isHexPrefixed.`,
+    );
+  }
+  return str.slice(0, 2) === '0x';
+}
```

Comment:
Is this method not exported from ethereumjs/util?

Reply:
It is. We can still use that version. I want us to use this version of `stripHexPrefix` for reasons that I will add to the PR description shortly here.

Reply:
`stripHexPrefix` has been extracted so that we can use the version that was used in `ethereumjs-util` `v7.1.0` - which comes from [strip-hex-prefix](https://npmfs.com/package/strip-hex-prefix/1.0.0/src/index.js#L8), via [ethjs-util](https://npmfs.com/package/ethjs-util/0.1.6/src/index.js#L2). This upgrade bumped us to the latest version of etherumjs-util (7.1.5) where a new implementation of `stripHexPrefix` is used, which breaks a bunch of assumptions on our end. Basically we pass non-string inputs to `stripHexPrefix` all over the place and the old implementation would just return them unaltered but [the new one throws an error](https://npmfs.com/package/ethereumjs-util/7.1.5/src/internal.ts#L46).

Reply:
I should have known, had similar issues with the v5->v7 upgrade back before EIP1559 for other utilities exported by this library.